Added python 3 support and cleaned up some testing files

henryrizzi opened the pull request:
I changed the library to use `response.text` rather than `response.content` as this will allow the text to be automatically decoded in both python 2.6+ and python 3. Also, the six library is used to bridge the gap between python 2 and 3.

haohany commented on ox3apiclient/__init__.py:
```diff
@@ -193,13 +194,13 @@ def authorize_token(self, email=None, password=None):
-        self._token['verifier'] = parse_qs(response.content)['oauth_verifier'][0]
-
+        self._token['verifier'] = parse_qs(response.text)['oauth_verifier'][0]
+          
```
trailing whitespace

haohany commented on tests/client.py:
```diff
@@ -33,28 +32,27 @@ def setUp(self):
+        with patch('ox3apiclient.requests.Session') as\
+        self.mock_requests_session,\
+        patch('ox3apiclient.Client.log_request') as\
+        self.mock_client_log_request:
```
python 2.6 don't support this syntax

henryrizzi commented:
The trailing whitespace has been purged from __init__.py
client.py has been fixed to not use any fancy 2.7 features context features

The client.py tests won't all run right now, but the delete test is still failing on master. On python 2.6, a few tests will fail because of differences in the way that failUnlessRaises works in different versions, but the actual api should still function.

haohany commented on .gitignore:
```diff
@@ -1,2 +1,6 @@
+ox3apiclient.egg-info/
+landing/*
+build/*
+dist/*
```
these folders look specific to your environment. Maybe you want to change your ~/.gitignore instead.

henryrizzi commented:
I just removed those folders from my .gitignore. the egg-info is jut the file generated in python3 when you run `python setup.py develop` and the other ones are for installing using python2 variants. landing was just specific to me, but I removed all 4 items from the .gitignore anyway.

marcsantiago commented:
How stable is this branch for production?  It looks a lot cleaner than the requests_oauthlib branch i'm currently using.

henryrizzi commented:
I've used it to get a lot of reports, but so far haven't had to do anything with deleting, posting or putting. For getting reports at least, it is very stable and I haven't run into any issues so far on python 3.5.2

haohany commented:
I use it mainly to create and delete objects in API. requests_oauthlib is merged into this branch and some bugs are fixed.

henryrizzi commented:
Has anyone else tested this for their use case using python 3?

openx-luis commented:
Superseded by https://github.com/openx/OX3-Python-API-Client/pull/33
